includes/Modules/Sign_In_With_Google.php

```diff
@@ -82,23 +140,58 @@ public function handle_google_auth() {
 			exit;
 		}
 
-		$client_id = $this->get_settings()->get()['clientID'];
-		$id_token  = $this->context->input()->filter( INPUT_POST, 'credential' );
+		$id_token = $this->context->input()->filter( INPUT_POST, 'credential' );
 		try {
-			$client  = new Google_Client( array( 'client_id' => $client_id ) );
-			$payload = $client->verifyIdToken( $id_token );
+			$payload = $this->client->verifyIdToken( $id_token );
+		} catch ( \Exception $e ) {
+			wp_safe_redirect( add_query_arg( 'error', 'google_auth_invalid_request', wp_login_url() ) );
+			exit;
+		}
 
-			if ( empty( $payload ) ) {
-				wp_safe_redirect( add_query_arg( 'error', 'google_auth_invalid_request', wp_login_url() ) );
-				exit;
-			}
+		if ( is_null( $payload ) || empty( $payload ) || ! array_key_exists( 'sub', $payload ) || empty( $payload['sub'] ) || ! array_key_exists( 'email', $payload ) || empty( $payload['email'] ) ) {
+			wp_safe_redirect( add_query_arg( 'error', 'google_auth_invalid_request', wp_login_url() ) );
+			exit;
+		}
 
-			// @TODO implement further flow using $payload in #9339.
+		$google_user_id    = $payload['sub'];
+		$google_user_email = $payload['email'];
```

Comment:
Although the data will be received from Google API through the lib, it is still an external source, and in this case data will be used for querying/storing value in the database, perhaps it won't hurt to sanitize the values
```suggestion
        $google_user_id    = sanitize_text_field( $payload['sub'] );
        $google_user_email = sanitize_email( $payload['email'] );
```

Reply:
No need to sanitize it, this data comes from a trusted source and should be used as is.